src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/InputAbbreviationAsWordInNameIgnore.java

```diff
@@ -0,0 +1,134 @@
+package com.puppycrawl.tools.checkstyle.checks.naming.abbreviationaswordinname;
+
+/*
+ * Config:
+ * allowedAbbreviationLength = 5
+ * allowedAbbreviations = NUMBER,MARAZMATIC,VARIABLE
+ * tokens = CLASS_DEF,VARIABLE_DEF,METHOD_DEF,ENUM_DEF,ENUM_CONSTANT_DEF,
+ *      PARAMETER_DEF,INTERFACE_DEF,ANNOTATION_DEF
+ * ignoreStatic = true
+ * ignoreFinal = true
+ * ignoreStaticFinal = true
+ */
+public class InputAbbreviationAsWordInNameIgnore {
+
+    abstract class InputAbbreviationAsWordInNameType {
+    }
+
+    abstract class NonAAAAbstractClassName {
+    }
+
+    abstract class FactoryWithBADNAme {
+    }
+
+    abstract class AbstractCLASSName {
+        abstract class NonAbstractInnerClass {
+        }
+    }
+
+    abstract class ClassFactory1 {
+        abstract class WellNamedFactory {
+        }
+    }
+
+    class NonAbstractClass1 {
+    }
+
+    class AbstractClass1 {
+    }
+
+    class Class1Factory1 {
+    }
+
+    abstract class AbstractClassName3 {
+        class AbstractINNERRClass { // violation
```

Comment:
please in inputs static-final classes to make it clear that we ignore modifiers on types.
```
/var/tmp$ cat Test.java 
class Test {
  static final class Inner {}

}
/var/tmp$ javac Test.java 
/var/tmp$ echo $?
0

```

Reply:
@wltan , please always reply "done" to review comment, to make me sure that addressed it and ready for review.
Just new commit, could mean "I did partial update and pushed changes to remote for backup and make sure CI is green, while I sleeping or afk"

Reply:
Sorry, I wanted to comment after checking the pushed changes but I forgot about it. I'll be more careful next time.